### PR TITLE
Add `Git` storage + SSH documentation

### LINKED ISSUE #4538: `Git` storage with `use_ssh` does not work with Prefect image
## Description

`Git` storage does not work when using `prefecthq/prefect:latest` image. 

When trying to retrieve a flow using ssh `Git` storage, you'll see a run time error `[Errno 2] No such file or directory: 'ssh': 'ssh'`.

This is my fault, I overlooked this when writing `Git` storage.

## Expected Behavior

The storage class works smoothly when ssh keys are mounted in to the container directly. 

An example of mounting ssh keys using the prefect image should also be documented.

## Reproduction

1. Create Flow
```python
import os
from prefect import Flow, Parameter, task
from prefect.run_configs import KubernetesRun
from prefect.storage import Git
@task(log_stdout=True)
def say_hello(name):
    print("Hello, {}!".format(name))
with Flow("Hello World") as flow:
    thename = Parameter("name")
    say_hello(thename)
storage = Git(
    repo="my/repo",
    flow_path="src/flows/hello_world.py",
    branch_name="my/branch-name",
    use_ssh=True,
)
storage.add_flow(flow) # the Flow object
storage.get_flow(flow.name) # try to load the flow from git storage
```
2. Publish Flow to GitHub
3. Run Docker Image
`$ docker run --rm -it --entrypoint sh -v path/to/flow:/src -v ~/.ssh:/root/.ssh prefecthq/prefect:latest`
4. Debug Flow
```python
# python /src/my_test_file.py
Traceback (most recent call last):
  File "/src/TestCloningRepoWithSSH.py", line 26, in <module>
    storage.get_flow(flow.name) # try to load the flow from git storage
  File "/usr/local/lib/python3.7/site-packages/prefect/storage/git.py", line 122, in get_flow
    clone_depth=self.clone_depth,
  File "/usr/local/lib/python3.7/site-packages/prefect/utilities/git.py", line 48, in __enter__
    source=self.git_clone_url, target=self.temp_dir.name, depth=self.clone_depth
  File "/usr/local/lib/python3.7/site-packages/dulwich/porcelain.py", line 476, in clone
    **kwargs
  File "/usr/local/lib/python3.7/site-packages/dulwich/porcelain.py", line 1559, in fetch
    fetch_result = client.fetch(path, r, progress=errstream.write, depth=depth)
  File "/usr/local/lib/python3.7/site-packages/dulwich/client.py", line 528, in fetch
    depth=depth,
  File "/usr/local/lib/python3.7/site-packages/dulwich/client.py", line 1009, in fetch_pack
    proto, can_read, stderr = self._connect(b"upload-pack", path)
  File "/usr/local/lib/python3.7/site-packages/dulwich/client.py", line 1659, in _connect
    self.host, argv, port=self.port, username=self.username, **kwargs
  File "/usr/local/lib/python3.7/site-packages/dulwich/client.py", line 1522, in run_command
    stderr=subprocess.PIPE,
  File "/usr/local/lib/python3.7/subprocess.py", line 800, in __init__
    restore_signals, start_new_session)
  File "/usr/local/lib/python3.7/subprocess.py", line 1551, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: 'ssh': 'ssh'
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/weakref.py", line 648, in _exitfunc
    f()
  File "/usr/local/lib/python3.7/weakref.py", line 572, in __call__
    return info.func(*info.args, **(info.kwargs or {}))
  File "/usr/local/lib/python3.7/tempfile.py", line 797, in _cleanup
    _shutil.rmtree(name)
  File "/usr/local/lib/python3.7/shutil.py", line 485, in rmtree
    onerror(os.lstat, path, sys.exc_info())
  File "/usr/local/lib/python3.7/shutil.py", line 483, in rmtree
    orig_st = os.lstat(path)
FileNotFoundError: [Errno 2] No such file or directory: '/tmp/tmp9goeajxv'
```

### PR DESCRIPTION
**Pull request by zangell44**
## Summary

Adds ssh agent to Dockerfile, allowing the use of ssh cloning when using the image for `Git` storage.

## Changes

- Adds ssh-agent to Dockerfile
- Minor documentation update

## Importance

Resolves #4538.

## Checklist

This PR:

- [x] adds new tests (if appropriate)
- [x] adds a change file in the `changes/` directory (if appropriate)
- [x] updates docstrings for any new functions or function arguments, including `docs/outline.toml` for API reference docs (if appropriate)

### CODE REVIEW

**Comment by TylerWanner** [comment]
I don't see a compelling reason to add ssh clients to our base Docker image and would oppose doing so until convinced otherwise.  Is there any way to just install at runtime when using git storage?  Putting ssh clients in all flow containers feels heavy, though I can't make a complete argument that it's preventatively insecure

**Comment by zangell44** [comment]
Because of security concerns, we've decided not to add `ssh-agent` to our base image.

I'll update this PR to provide documentation on adding `ssh-agent` to the base image and configuring ssh keys.

File: docs/orchestration/flow_config/storage.md [hunk 1]
```diff
@@ -515,3 +520,77 @@ Template strings in `${}` are used to reference sensitive information. Given
...
+
+When using a [Kubernetes agent](/orchestration/agents/kubernetes.html#kubernetes-agent), SSH keys can be mounted as secret volumes.
+
+First, create a [Kubernetes Secret](https://kubernetes.io/docs/concepts/configuration/secret/) containing our SSH key and known hosts file.
+
+```bash
+kubectl create secret generic my-ssh-key --from-file=<ssh-key-name>=/path/to/<ssh-key-name> --from-file=known_hosts=/path/to/known_hosts
+```
+
+Next, create a custom job template to mount the secret volume to `/root/.ssh`.
```
Comment (TylerWanner) [inline docs/orchestration/flow_config/storage.md:571]: It's kinda weird that the example is not a full "job template" even though it is minimally viable to be properly merged with the default job template.  this story continues to haunt me but this is good imo

File: docs/orchestration/flow_config/storage.md [hunk 2]
```diff
@@ -515,3 +520,77 @@ Template strings in `${}` are used to reference sensitive information. Given
...
+          secret:
+            secretName: my-ssh-key
+            optional: false
+            defaultMode: 0600
+```
+
+Finally, [configure the agent or flow to use the custom job template](https://docs.prefect.io/orchestration/agents/kubernetes.html#custom-job-template). 
+
+
+Creating a [Kubernetes service account](https://kubernetes.io/docs/reference/access-authn-authz/service-accounts-admin/) to permission the Secret properly is recommended. Once configured in Kubernetes, service account can be set either [on agent start or on the run config](https://docs.prefect.io/orchestration/agents/kubernetes.html#service-account).
```
Comment (TylerWanner) [inline docs/orchestration/flow_config/storage.md:596]: I believe the RBAC config is a must for this to work

**Comment by TylerWanner** [comment]
New docs look good.  Definitely an improvement

**Review by TylerWanner — APPROVED** [review]

File: docs/orchestration/flow_config/storage.md (hunk 2 above)
Comment (madkinsz) [reply docs/orchestration/flow_config/storage.md:596]: @zangell44 Does this need a change before merge?

Comment (zangell44) [reply docs/orchestration/flow_config/storage.md:596]: @madkinsz this is okay to merge as is, I've followed these instructions successfully myself
I have my eye on this documentation to update it as we get feedback

**Pull request merged**